run/events_pubsub/main.go

```diff
@@ -12,38 +12,49 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// [START run_events_pubsub_server]
+// [START run_events_pubsub_handler]
 
-// Sample run-pubsub-events is a Cloud Run service which handles Pub/Sub messages.
+// Sample run-events-pubsub is a Cloud Run service which handles Pub/Sub messages.
 package main
 
 import (
-	"context"
+	b64 "encoding/base64"
+	"encoding/json"
 	"fmt"
 	"log"
 	"net/http"
 	"os"
-
-	cloudevents "github.com/cloudevents/sdk-go/v2"
 )
 
-var (
-	handler = http.DefaultServeMux
-)
+// PubSubMessage is the payload of a Pub/Sub event.
+type PubSubMessage struct {
+	Message struct {
+		Data string `json:"data,omitempty"`
```

Comment:
Should this be `[]byte` instead of `string`? Would be able to forget about `base64` since it should be handled automatically.

Reply:
In this case, we really want a string since the HTTP request sends a JSON string for the value. Otherwise we'll just need to convert the `[]byte` to a string later right?

LMK if there's a reason to use a `[]byte`.

Reply:
Nope. Below, you do `nameBytes, err := base64.StdEncoding.DecodeString(e.Message.Data)`. You could delete all of that and just do:

```
name := string(e.Message.Data)
```

if `Data` was of type `[]byte`.

Reply:
Interesting. Fixed.

I guess a byte array to string will base 64 decode it?